freedomofpress/securedrop
# Ensure python tooling requires hashes, bump builders trusty->xenial

This change helps maintain that each developer is pulling the exact same
resources, help mitigate against nefarious actors in the pip pipeline,
and solve male baldness.

## Status

Work in Progress

## Description of Changes

Kinda, sorta addresses #3270 .
Doesn't do anything to squash the multiple hashes for some packages tho.

Changes proposed in this pull request:

* Breaks out pip update make target into a script
* Adds hashes and removes the banner on those generated requirements files.
* Replaces trusty with xenial for the builder containers. Why did I do this? Well I encountered a fun chicken before the egg problem with pip. Specifically I wanted to utilize hashes but the pip version installed in trusty does not support hashes (its really old, like 1.x series). It would have been super silly to peg all hashes and then just wildly run `pip install -U pip` before that. Sooooo I made the call to jump the builders to xenial which has a much newer version of pip pre-installed.
* Fixes some misc ansible errors

On a side note, our quay Clare scanner reported dramatically less vulnerabilities when using xenial vs. trusty. Take that report with a grain of salt but hey i consider this a win.

## Testing

How should the reviewer test this PR?

* Make sure CI passes :+1: 
* Make sure you can build debs and that `vagrant up /staging/` comes up.


## Deployment

Any special considerations for deployment?

This doesn't dramatically change deployment story but it does update a lot of the tooling behind the builder.
